pageserver/src/tenant/timeline.rs

```diff
@@ -1907,6 +1907,8 @@ impl Timeline {
 
             true
         } else if projected_layer_size >= checkpoint_distance {
+            // NB: The InMemoryLayerIndexValue::does_timeline_should_roll_prevent_overflow relies on us
```

Comment:
`InMemoryLayerIndexValue::does_timeline_should_roll_prevent_overflow` doesn't exist. Did you mean `InMemoryLayerIndexValue::does_timeline_should_roll_prevent_failure`? Even the, I still don't understand the comment.